### PR TITLE
Add resample list

### PR DESCRIPTION
**Pull request by kfkaplan**
Added a definition in echelle.py called `resample_list`.  I created it because I wanted to resample a synthetic standard star spectrum (for example as done in this tutorial:  https://github.com/OttoStruve/muler/blob/main/docs/tutorials/telluric_correction_with_gollum.ipynb) onto the same wavelength grid and with the same multiple orders as my science spectrum for use in relative flux calibration and telluric correction.

I tried various alternative approaches with this as a method for the EchelleSpectrum and EchelleSpectrumList classes but settled on making it its own definition independent of the classes since any synthetic spectrum is likely going to be a specutils Spectrum object, and not  EchelleSpectrum or EchelleSpectrumList.  Alternative ideas are welcome, if anyone can think of a better approach.  Otherwise, this has been tested and seems to work.

### CODE REVIEW

**Comment by gully** [comment]
Interesting idea, thanks for the contribution!

Have you considered an approach using `.stitch()`?  That is, stitching the data spectrum into a single (with gaps) spectrum, rather than covering the model to a SpectrumList-like object with the same shape as data SpectrumList?  How do you think most people would think about a model resampled in this way?  Some echelle spectra---like IGRINS--- have overlapping orders, and so resampling to those orders makes sense.  

Interesting contribution, it really made me think about some different scenarios!  

The tests are failing because of a recent change.  I'm going to merge #44, can you merge those changes into here in a minute...

**Review by gully — COMMENTED** [review]
Would you be open to adding a unit test?  I could help walk you through it

File: src/muler/echelle.py [hunk 1]
```diff
@@ -62,6 +62,18 @@
     from specutils import SpectrumList
 
 
+def resample_list(spec_to_resample, specList, **kwargs):
+    """
+    Resample a single EchelleSpectrum or Spectrum1D object into a EchelleSpectrumList object.
+    Useful for converting models into echelle spectra with multiple orders.
+    """
```
[the remark this answers is not in the document]
Comment (kfkaplan) [reply src/muler/echelle.py:69]: Updated description and moved to `src/utilities.py`.  I didn't realize that module was there.

[the remark this answers is not in the document]
Comment (kfkaplan) [reply src/muler/echelle.py:69]: A unit test would be nice but I have not done it before.  How would I go about doing this?  Is it much work?

**Comment by kfkaplan** [comment]
> Interesting idea, thanks for the contribution!
> 
> Have you considered an approach using `.stitch()`? That is, stitching the data spectrum into a single (with gaps) spectrum, rather than covering the model to a SpectrumList-like object with the same shape as data SpectrumList? How do you think most people would think about a model resampled in this way? Some echelle spectra---like IGRINS--- have overlapping orders, and so resampling to those orders makes sense.
>
> Interesting contribution, it really made me think about some different scenarios!
> 

I created this definition to resample synthetic standard star spectra to the spectrum of an observed standard star spectrum for purposes of flux calibration and telluric correction.  The synthetic spectrum lies on a completely different wavelength grid.  The goal here is to do most of the processing on the observed data before resampling the different orders into a single spectrum, in order to avoid any unintended systematics and/or algorithmic noise that might be introduced when resampling data.  The final data product could then be resampled into a single spectrum.  Alternatively, someone could use `.stitch()` on the list and just need to run the regular `resample` on the synthetic spectrum and any systematics+noise introduced would be too small to care about.  I would like to leave it up to the user to decide what they want to do.

> The tests are failing because of a recent change. I'm going to merge #44, can you merge those changes into here in a minute...

I merged the latest version of main with this branch.  Looks like it fixed the problem.

**Review by gully — APPROVED** [review]
Awesome, these changes look great, and I am happy that the resample list now resides in the utilities module, which seems like the right place for it.  The rv_shift function will be useful.
Thanks again @kfkaplan !

File: src/muler/echelle.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gully) [reply src/muler/echelle.py]: Let's go ahead and merge this nice Pull Request and we can add some unit tests later.  I may add some developer documentation about adding tests, and happy to walk through it at that point.  Thanks again!

**Pull request merged**